google/resource_storage_default_object_acl.go

```diff
@@ -0,0 +1,199 @@
+package google
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"google.golang.org/api/storage/v1"
+)
+
+func resourceStorageDefaultObjectAcl() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceStorageDefaultObjectAclCreate,
+		Read:   resourceStorageDefaultObjectAclRead,
+		Update: resourceStorageDefaultObjectAclUpdate,
+		Delete: resourceStorageDefaultObjectAclDelete,
+
+		Schema: map[string]*schema.Schema{
+			"bucket": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"role_entity": &schema.Schema{
+				Type:     schema.TypeList,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func getDefaultObjectAclId(bucket string) string {
+	return bucket + "-default-acl"
```

Comment:
should we make it bucket + "-default-object-acl"?

Reply:
Merged upstream into the fork.

I tried to make it consistent with object_acl resource, where they ID object + "-acl" but since it's a bucket level change, I think you are right it will be, more staright forward if we add object in there. changing to "-default-object-acl".

Logging seems consistent to me. Added a small change